src/main/java/backend/resource/MultiModel.java

```diff
@@ -13,6 +13,9 @@
 import java.util.stream.Collectors;
 
 /**
+ * This represents the true state of local repositories data. Operations meant to change the state
+ * of local repositories data but do not go through the methods in this class i.e. operating on
+ * dangling references its sub-components are considered unsafe
```

Comment:
We change labels by calling `models.replaceIssueLabels` and passing in the issue's id, the repo's id and the list of labels and no dangly object is modified so I guess that fits into this criteria.